lld/COFF/Driver.cpp

```diff
@@ -2307,8 +2307,11 @@ void LinkerDriver::linkerMain(ArrayRef<const char *> argsArr) {
 
   // Handle /dependentloadflag
   for (auto *arg :
-       args.filtered(OPT_dependentloadflag, OPT_dependentloadflag_opt))
+       args.filtered(OPT_dependentloadflag, OPT_dependentloadflag_opt)) {
     parseDependentLoadFlags(arg);
+    if (!ctx.symtab.findUnderscore("_load_config_used"))
+      warn("_load_config_used not found, /delayloadflag will have no effect");
+  }
```

Comment:
Is it right to check for the symbol so early? I thought at this point the objects/libraries haven't even been loaded yet so the symbol will always be missing.

I think the check should be added here instead: https://github.com/llvm/llvm-project/blob/3a31427224d4fa49d7ef737b21f6027dc4928ecf/lld/COFF/Writer.cpp#L2604-L2611
It can probably just check for a non-zero value, like in https://github.com/llvm/llvm-project/blob/3a31427224d4fa49d7ef737b21f6027dc4928ecf/lld/COFF/Writer.cpp#L2634-L2635

Reply:
I had started there, but judging by other uses of `ctx.symtab` I figured the driver might be the right place to put it.

Reply:
My preference would be to put the check closer to where the field in `_load_config_used` is being set, and given that `Writer::prepareLoadConfig` already does other checks on `_load_config_used`, it just seems to me to be the more logical place to add the check.

(I was also a bit concerned that resolving the symbol earlier than before may subtly change the behaviour and output of LLD, but I guess there isn't really much to break here.)

Reply:
In addition, the current code would emit one warning per every `/dependentloadflag` flag specified (when multiple are passed, only the last one would take effect).

Also, I just noticed the typo &ndash; the warning message refers to "/delayloadflag" instead of `/dependentloadflag`.

Reply:
> My preference would be to put the check closer to where the field in `_load_config_used` is being set, and given that `Writer::prepareLoadConfig` already does other checks on `_load_config_used`, it just seems to me to be the more logical place to add the check.

Sure, I don't have a strong option here for either approach.

> In addition, the current code would emit one warning per every `/dependentloadflag` flag specified (when multiple are passed, only the last one would take effect).

Oh, that's right. I'll see what I can do about it.

> Also, I just noticed the typo – the warning message refers to "/delayloadflag" instead of `/dependentloadflag`.

Dang, good catch!